### PR TITLE
Dash disk

### PR DESCRIPTION
**Pull request by ttgenius**
Added plots and texts for showing hpc quota usage
[2019-03-21 12:47:42,350] ERROR in app: Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.6/site-packages/dash/dash.py", line 1059, in dispatch
    for component_registration in self.callback_map[output]['inputs']:
KeyError: 'maui_daily_quota.figure'

### CODE REVIEW

File: dashboard/DashboardDB.py [hunk 1]
```diff
@@ -208,6 +222,77 @@ def get_squeue_entries(self, hpc: const.HPC):
 
         return [SQueueEntry(*result) for result in results]
 
+    def update_daily_quota(self, entries: Iterable[QuotaEntry], hpc: const.HPC):
```
Comment (claudio525) [inline dashboard/DashboardDB.py]: Docstring

File: dashboard/DashboardDB.py [hunk 2]
```diff
@@ -227,6 +312,21 @@ def _create_queue_table(self, cursor, hpc: const.HPC):
             )
         )
 
+    def _create_quota_table(self, cursor, hpc: const.HPC):
```
Comment (claudio525) [inline dashboard/DashboardDB.py]: No need to make this a function as it is only ever called from a single location

File: dashboard/run_data_collection.py [hunk 3]
```diff
@@ -115,6 +121,12 @@ def collect_data(self, hpc: const.HPC):
                         None,
                     ),
                 )
+        # get quota
+        quota_output = self.run_cmd(
+            hpc.value, "nn_check_quota | grep 'nesi00213'", timeout=60
```
Comment (claudio525) [inline dashboard/run_data_collection.py]: This cmd should be fast, so leave the timeout to its default value unless it starts timing out.

File: dashboard/DashboardDB.py [hunk 4]
```diff
@@ -208,6 +222,77 @@ def get_squeue_entries(self, hpc: const.HPC):
...
+        return results
+
+    def get_daily_quota(
+        self,
+        hpc: const.HPC,
+        day: Union[date, str] = None,
+        file_system="nobackup",
+    ):
+        """
+        Get daily quota usage for a particular file system eg.nobackup/project
```
Comment (claudio525) [inline dashboard/DashboardDB.py]: Single line docstrings should be of the format 
"""Single line Docstring"""  

and multi line docstrings
"""First line
Second line
"""

File: dashboard/DashboardDB.py (hunk 1 above)
Comment (ttgenius) [reply dashboard/DashboardDB.py]: all updated as per comment

**Pull request merged**